src/sensors/character_collision_sensor.py

```diff
@@ -14,4 +14,5 @@ def get_reading(self, position):
             return self.player
         for npc in self.npcs:
             if position.distance_to(npc.pos) < self.radius["basic_npc"]:
-                return npc
\ No newline at end of file
+                return npc
```

Comment:
You should check for equal IDs here, otherwise for the first NPC the sensor will always return that first NPC and no collision would be detected. Probably you will need a character_id property inside the sensor, and create a new sensor for each NPC (and player)